src/components/SideBar.tsx

```diff
@@ -260,6 +262,12 @@ export default function SideBar({ mobile, closeSidebar }: { mobile?: boolean; cl
         return imported
     }, [chainId])
 
+    const onTabClick = (e: any) => {
+        const key = e.target.getAttribute('data-key')
+        mobile && closeSidebar()
+        sendData({ event: 'goto_page', action: `goto_${key}` })
```

Comment:
better to subscribe to navigation component and listen for current page(route) change there.

this will handle all the cases we're navigating (e.g. redirects, manual link change in the address line etc)

Reply:
Moved the events for navigation within the app to App.tsx
Kept this setup (new method onExternalClick) for external links